code/modules/xenoarchaeology/traits/xenoartifact_majors.dm

```diff
@@ -0,0 +1,589 @@
+	flags = URANIUM_TRAIT
+	weight = 25 //annoying trait
+
+/datum/xenoartifact_trait/major/emp/activate(obj/item/xenoartifact/X)
+	playsound(get_turf(X), 'sound/magic/disable_tech.ogg', 50, TRUE)
+	empulse(get_turf(X.loc), max(1, X.charge*0.03), max(1, X.charge*0.07, 1)) //This might be too big
+
+///============
+/// Invisible, makes the target invisible for a short time
+///============
+/datum/xenoartifact_trait/major/invisible //One step closer to the one ring
+	label_name = "Transparent"
+	label_desc = "Transparent: The shape of the Artifact is difficult to percieve. You feel the need to call it, precious..."
+	weight = 25
+	var/list/victims = list()
+
+/datum/xenoartifact_trait/major/invisible/on_item(obj/item/xenoartifact/X, atom/user, atom/item)
+	if(istype(item, /obj/item/laser_pointer))
+		var/obj/item/laser_pointer/L = item
+		if(L.energy)
+			to_chat(user, "<span class='info'>The [item.name]'s light passes through the structure.</span>")
+			return TRUE
+	..()
+
+/datum/xenoartifact_trait/major/invisible/activate(obj/item/xenoartifact/X, mob/living/target)
+	if(isliving(target))
+		victims += target
+		hide(target)
+		addtimer(CALLBACK(src, .proc/reveal, target), ((X.charge*0.4) SECONDS))
+		X.cooldownmod = ((X.charge*0.4)+1) SECONDS
+
+/datum/xenoartifact_trait/major/invisible/proc/hide(mob/living/target)
+	ADD_TRAIT(target, TRAIT_PACIFISM, type)
+	animate(target, ,alpha = 0, time = 5)
+
+/datum/xenoartifact_trait/major/invisible/proc/reveal(mob/living/target)
+	if(target)
+		REMOVE_TRAIT(target, TRAIT_PACIFISM, type)
+		animate(target, ,alpha = 255, time = 5)
+		target = null
+
+/datum/xenoartifact_trait/major/invisible/Destroy()
+	. = ..()
+	for(var/mob/living/M in victims)
+		reveal(M)
+
+///============
+/// Teleports the target to a random nearby location
+///============
+/datum/xenoartifact_trait/major/teleporting
+	desc = "Displaced"
+	label_desc = "Displaced: The shape's state is unstable, causing it to shift through planes at a localized axis. Just ask someone from science..."
+	flags = BLUESPACE_TRAIT | PLASMA_TRAIT | URANIUM_TRAIT
+
+/datum/xenoartifact_trait/major/teleporting/activate(obj/item/xenoartifact/X, atom/target, atom/user)
+	if(istype(target, /atom/movable))
+		var/atom/movable/victim = target //typecast to access anchored
+		if(!victim.anchored)
+			do_teleport(victim, get_turf(victim), (X.charge*0.3)+1, channel = TELEPORT_CHANNEL_BLUESPACE)
+
+///============
+/// Lamp, creates a light with random color for a short time
+///============
+/datum/xenoartifact_trait/major/lamp
+	label_name = "Lamp"
+	label_desc = "Lamp: The Artifact emits light. Nothing in its shape suggests this."
+	flags = BLUESPACE_TRAIT | URANIUM_TRAIT
+	var/light_mod
+
+/datum/xenoartifact_trait/major/lamp/on_init(obj/item/xenoartifact/X)
+	X.light_system = MOVABLE_LIGHT
+	X.light_color = pick(LIGHT_COLOR_FIRE, LIGHT_COLOR_BLUE, LIGHT_COLOR_GREEN, LIGHT_COLOR_RED, LIGHT_COLOR_ORANGE, LIGHT_COLOR_PINK)
```

Comment:
This should also use the lighting interface things (X.set_light_color or whatever its called)